lib/task_helpers.rb

```diff
@@ -1,32 +1,64 @@
+require 'elasticsearch'
+
 module TaskHelpers
   module ElasticsearchHelper
-    def self.create_index(name=nil)
-      name ||= "#{Content::ES_INDEX_NAME}_#{Time.now.strftime('%Y%m%d%H%M%S')}"
-      index = Tire.index(name)
+    LOG = Logger.new(STDERR)
 
-      LOG.info "Creating new index: #{name}..."
-      index.create
+    def self.create_index(name=nil)
+      name ||= "#{Content::ES_INDEX_NAME}_#{Time.now.strftime('%Y%m%d%H%M%S%L')}"
 
-      [CommentThread, Comment].each do |model|
-        LOG.info "Applying index mappings for #{model.name}"
-        model.put_search_index_mapping(index)
+      mappings = {}
+      [Comment, CommentThread].each do |model|
+        mappings.merge! model.mappings.to_hash
       end
-      LOG.info '...done!'
 
-      index
+      Elasticsearch::Model.client.indices.create(index: name, body: {mappings: mappings})
+      LOG.info "Created new index: #{name}."
+      name
     end
 
     def self.delete_index(name)
-      Tire.index(name).delete
+      begin
+        Elasticsearch::Model.client.indices.delete(index: name)
+        LOG.info "Deleted index: #{name}."
+      rescue Elasticsearch::Transport::Transport::Errors::NotFound
+        # NOTE (CCB): Future versions of the Elasticsearch client support the ignore parameter,
+        # that can be used to ignore 404 errors.
+        LOG.info "Unable to delete non-existent index: #{name}."
+      end
     end
 
-    def self.get_index
-      CommentThread.tire.index
+    def self.get_index_shard_count(name)
+      settings = Elasticsearch::Model.client.indices.get_settings(index: name)
+      settings[name]['settings']['index']['number_of_shards']
     end
 
-    def self.get_index_shard_count(name)
-      settings = Tire.index(name)
-      settings['index.number_of_shards']
+    def self.exists_alias(alias_name)
+      Elasticsearch::Model.client.indices.exists_alias(name: alias_name)
+    end
+
+    def self.move_alias(alias_name, index_name)
+      actions = [
+          {add: {index: index_name, alias: alias_name}}
+      ]
+
+      begin
+        response = Elasticsearch::Model.client.indices.get_alias(name: alias_name)
+        if response.length
+          actions.unshift({remove: {index: response.keys.join(','), alias: alias_name}})
```

Comment:
FYI: I added code to `move_alias` that allows someone to recover from this issue if it ever happens.